internal/pkg/ecs/ecs.go

```diff
@@ -467,27 +467,31 @@ func (c Client) stateMachineARN(app, env, job string) (string, error) {
 	return stateMachineARN, nil
 }
 
-// NonZeroExitCode returns the non zero exit code of the task if any.
-func (c Client) NonZeroExitCode(taskARNs []string, groupName string, cluster string) error {
-	essentialContainers := make(map[string]bool)
-	taskDefName := fmt.Sprintf("copilot-%s", groupName)
-	taskDefinition, err := c.ecsClient.TaskDefinition(taskDefName)
+// HasNonZeroExitCode returns an error if at least one of the tasks exited with a non-zero exit code. It assumes that all tasks are built on the same task definition.
+func (c Client) HasNonZeroExitCode(taskARNs []string, cluster string) error {
+	tasks, err := c.ecsClient.DescribeTasks(cluster, taskARNs)
 	if err != nil {
-		return fmt.Errorf("get task definition %s: %w", taskDefName, err)
+		return err
```

Comment:
I think we can probably wrap it here:
```suggestion
		return fmt.Errorf("describe tasks %s: err", taskARNs, err)
```